plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/p3order/LayerSweepCrossingMinimizer.java

```diff
@@ -206,9 +206,9 @@ private int minimizeCrossingsWithCounter(final GraphInfoHolder gData) {
             gData.crossMinimizer().setFirstLayerOrder(gData.currentNodeOrder(), isForwardSweep);
         } else {
             isForwardSweep = true;
-            gData.lGraph().setProperty(InternalProperties.FIRST_TRY_WITH_INITIAL_ORDER, false);
```

Comment:
Why did you have to move it outside?

Reply:
To be able to access it in `sweepReducingCrossings` to prevent random permutation of sinks that are not in the first layer (which is done by the barycenter heuristic).